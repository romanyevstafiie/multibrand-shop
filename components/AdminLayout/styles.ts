import styled from 'styled-components';

export const LayoutContainer = styled.section`
  display: grid;
  grid-template-columns: 1fr 8fr;
  min-height: 100vh;
`;

export const ChildrenWrapper = styled.main`
  background-color: ${({ theme }) => theme.dashboardGrayBg};
`;
