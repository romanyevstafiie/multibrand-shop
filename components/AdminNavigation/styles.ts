import styled from 'styled-components';

export const StyledAdminNavigation = styled.aside`
  display: flex;
  align-items: center;
  justify-content: center;
  position: sticky;
  top: 0;
  padding: 5rem 1rem;

  li {
    margin: 1rem 0;
  }
`;
