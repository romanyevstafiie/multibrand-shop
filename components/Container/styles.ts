import styled from 'styled-components';

export const StyledContainer = styled.section`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  max-width: ${({ theme }) => theme.maxWidth};
  margin: 0 auto;
  padding: 0 1rem;

  a {
    text-decoration: none;
  }
`;
