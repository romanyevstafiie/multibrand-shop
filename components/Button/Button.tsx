import React, { FC, ReactNode } from 'react';
import { StyledButton } from './styles';

export interface Props {
  isLoading?: boolean;
  isCompleted?: boolean;
  disabled?: boolean;
  text?: string;
  children?: ReactNode;
}

/**
 * Reusable generic button
 *@function Button
 *@param {boolean} isLoading - set styles for loading state
 *@param {boolean} isCompleted - set styles for completed state
 *@param {boolean} disabled - set styles for disabled state
 *@param {string} text - set button text
 *@param {children} ReactNode - set button children
 *@returns {JSX.Element} - rendered button
 */
const Button: FC<
  React.DetailedHTMLProps<
    React.ButtonHTMLAttributes<HTMLButtonElement>,
    HTMLButtonElement
  > &
    Props
> = ({ children, isLoading, isCompleted, text, ...rest }: Props) => {
  return (
    <StyledButton
      isLoading={isLoading}
      isCompleted={isCompleted}
      {...rest}
      className="btn"
    >
      {children}
      {text && text}
    </StyledButton>
  );
};

export default Button;
