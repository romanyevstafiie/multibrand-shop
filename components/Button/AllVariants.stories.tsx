import React from 'react';
import Container from '../Container/Container';
import { LoadingButton, Primary } from './Button.stories';
export default {
  title: 'test/AllVariants',
  decorators: [(story) => <Container>{story()}</Container>],
};

export const AllVariants = (): JSX.Element => {
  return (
    <>
      <Primary
        text="hello"
        disabled={false}
        isLoading={false}
        isCompleted={false}
      />
      <LoadingButton disabled={false} isLoading={true} isCompleted={false} />
    </>
  );
};
