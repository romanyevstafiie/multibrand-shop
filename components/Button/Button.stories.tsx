import { Story } from '@storybook/react';
import React from 'react';
import Container from '../Container/Container';
import TestButton, { Props } from './Button';

export default {
  title: 'test/FlexibleButton',
  component: TestButton,
  // args: {
  //   children: 'Button',
  // },
  decorators: [(story) => <Container>{story()}</Container>],
};

const Template: Story<Props> = (args) => <TestButton {...args} />;

export const Primary = Template.bind({});
Primary.args = { children: 'Primary' };
export const LoadingButton = Template.bind({});
LoadingButton.args = { isLoading: true, text: 'Loading' };
export const DisabledButton = Template.bind({});
DisabledButton.args = { disabled: true, text: 'Disabled' };
export const CompletedButton = Template.bind({});
CompletedButton.args = { isCompleted: true, text: 'Completed ü•≥' };

// extend args
export const ExtendArgs = Template.bind({});
ExtendArgs.args = {
  ...Primary.args,
  text: '‚≠ê',
};
