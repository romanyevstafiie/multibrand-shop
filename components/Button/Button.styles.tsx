import styled, { css, keyframes } from 'styled-components';
import { Props } from './Button';

/***
 * DEFAULT BUTTON STATE
 */
const defaultButton = css`
  background: ${({ theme }) => theme.primaryGolden};
`;
/***
 * HOVERED BUTTON STATE
 */
const hoveredButton = css`
  opacity: 0.95;
  transform: translateY(-2px);
  transition: transform 250ms cubic-bezier(0.3, 0.7, 0.4, 1.5);
`;
/***
 * FOCUSED BUTTON STATE
 */
const focusedButton = css`
  &:focus-visible {
    box-shadow: 0 0 0 3px ${({ theme }) => theme.secondaryBg};
  }
`;
/***
 * DISABLED BUTTON STATE
 */
const disabledButton = css`
  background: ${({ theme }) => theme.darkGray};
  cursor: not-allowed;
`;
/***
 * ACTIVE BUTTON STATE
 */
const activeButton = css`
  transform: scale(0.97);

  :before {
    display: none;
  }

  :after {
    content: '';
    position: absolute;
    z-index: 1;
    bottom: 1px;
    left: 1px;
    right: 1px;
    height: 55%;
    background: linear-gradient(
      rgba(255, 255, 255, 0),
      rgba(255, 255, 255, 0.75)
    );
    border-radius: 14px 14px 14px 14px/ 4px 4px 100px 100px;
  }
`;

const loadingKeyframes = keyframes`
  0% {
    transform: translateX(25px);
  }
  100% {
    transform: translateX(-20px);
  }
`;
/***
 * LOADING BUTTON STATE
 */
const movingGradient = css`repeating-linear-gradient(60deg,
#ffa500,
#ffa500 10px,
${({ theme }) => theme.primaryGolden} 10px,
${({ theme }) => theme.primaryGolden} 20px)
`;

const loadingButton = css`
  &::before {
    content: '';
    position: absolute;
    z-index: 1;
    top: 0;
    left: -100%;
    width: 300%;
    height: 100%;
    background: ${({ theme }) => theme.secondaryBg} ${movingGradient};
    animation: ${loadingKeyframes} 2s infinite linear;
  }

  & > span {
    opacity: 0.5;
  }

  cursor: wait;
`;
/***
 * COMPLETED BUTTON STATEAir
 */
const completedButton = css`
  background: ${({ theme }) => theme.primaryGreen};
  cursor: default;
`;

/***
 * BUTTON BASE STYLES
 */
export const BTN = styled.button<Props>`
  position: relative;
  overflow: hidden;

  display: inline-flex;
  align-items: center;
  justify-content: center;

  padding: 0.75rem 1rem;

  font-family: inherit;
  font-size: 1rem;
  text-transform: uppercase;
  font-weight: 600;
  background: transparent;
  border: 0;
  border-radius: ${({ theme }) => theme.borderRadius};
  color: ${({ theme }) => theme.primaryWhite};
  cursor: pointer;
  outline: none;

  text-shadow: rgb(0 0 0 / 15%) 1px 1px 1px;
  transition: 0.3s all ease;

  :before {
    content: '';
    position: absolute;
    z-index: 1;
    top: 1px;
    left: 2px;
    right: 2px;
    height: 35%;
    background: linear-gradient(rgba(255, 255, 255, 1), rgba(255, 255, 255, 0));
    border-radius: 4px 4px 100px 100px / 14px 14px 14px 14px;
  }

  ${(p) => !p.isLoading && defaultButton};

  &:hover {
    ${(p) => !p.disabled && !p.isLoading && !p.isCompleted && hoveredButton};
  }

  &:focus {
    ${(p) => !p.disabled && !p.isLoading && !p.isCompleted && focusedButton};
  }

  &:active {
    ${(p) => !p.disabled && !p.isLoading && !p.isCompleted && activeButton};
  }

  ${(p) => p.isLoading && loadingButton};

  &:disabled {
    ${disabledButton};
  }

  ${(p) => p.isCompleted && completedButton};

  @media ${({ theme }) => theme.media.phone} {
    font-size: 0.75rem;
  }
`;
