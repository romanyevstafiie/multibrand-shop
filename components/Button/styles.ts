import styled, { css, keyframes } from 'styled-components';
import { Props } from './Button';

const defaultButton = css`
  background: ${({ theme }) => theme.primaryGolden};
`;

const hoveredButton = css`
  background: ${({ theme }) => theme.secondaryGolden};
`;

const focusedButton = css`
  box-shadow: 0 0 0 3px ${({ theme }) => theme.secondaryGolden};
`;

const disabledButton = css`
  background: ${({ theme }) => theme.darkGray};
  cursor: not-allowed;
`;

const loadingKeyframes = keyframes`
  0% {
    transform: translateX(25px);
  }
  100% {
    transform: translateX(-20px);
  }
`;

const loadingButton = css`
  &::before {
    content: '';
    position: absolute;
    z-index: 1;
    top: 0;
    left: -100%;
    width: 300%;
    height: 100%;
    background: ${({ theme }) => theme.primaryGolden}
      repeating-linear-gradient(
        60deg,
        transparent,
        transparent 10px,
        #ffa500 10px,
        #ffc130 20px
      );
    animation: ${loadingKeyframes} 1s infinite linear;
  }
  & > span {
    opacity: 0.5;
  }
  cursor: wait;
`;

const completedButton = css`
  padding: 0 2rem;
  background: ${({ theme }) => theme.primaryGreen};
  cursor: default;
`;

export const StyledButton = styled.button<Props>`
  position: relative;
  overflow: hidden;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  height: 3rem;
  padding: 0 2rem;
  font-family: inherit;
  font-size: 1rem;
  text-transform: uppercase;
  font-weight: 700;
  background: transparent;
  border: 0;
  border-radius: ${({ theme }) => theme.borderRadius};
  color: ${({ theme }) =>
    theme.mode === 'dark' ? theme.primaryWhite : theme.secondaryBg};
  cursor: pointer;
  outline: none;
  & > span {
    position: relative;
    display: flex;
    align-items: center;
    z-index: 1;
  }
  ${(p) => !p.isLoading && defaultButton};
  &:hover {
    ${(p) => !p.disabled && !p.isLoading && !p.isCompleted && hoveredButton};
  }
  &:focus {
    ${(p) => !p.disabled && !p.isLoading && !p.isCompleted && focusedButton};
  }
  &:disabled {
    ${disabledButton};
  }
  ${(p) => p.isLoading && loadingButton};
  ${(p) => p.isCompleted && completedButton};
`;
