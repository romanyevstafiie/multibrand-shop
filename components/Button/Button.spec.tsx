import { mount } from 'enzyme';
import 'jest-styled-components';
import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import { mainTheme } from '../../styles/mainTheme';
import { withReduxAndStyledProviders } from '../../test/testUtils';
import Button from './Button';

/**
 * Setup function for the component
 * @returns {ShallowWrapper}
 */
const setup = () => {
  return mount(
    withReduxAndStyledProviders(
      <Button
        text="tomato"
        isCompleted={false}
        isLoading={true}
        disabled={false}
      />
    )
  );
};

describe('Button', () => {
  const wrap = setup();

  it('should render text content without throwing an error', () => {
    expect(wrap);
    expect(wrap.find('button').text()).toBe('tomato');
  });

  test('it applies default styles', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={mainTheme}>
          <Button />
        </ThemeProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule('border', '0');
    expect(tree).toHaveStyleRule('cursor', 'pointer');
  });
});
