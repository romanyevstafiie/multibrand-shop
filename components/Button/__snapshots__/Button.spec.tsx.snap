// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button it applies default styles 1`] = `
.c0 {
  position: relative;
  overflow: hidden;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 3rem;
  padding: 0 2rem;
  font-family: inherit;
  font-size: 1rem;
  text-transform: uppercase;
  font-weight: 700;
  background: transparent;
  border: 0;
  cursor: pointer;
  outline: none;
}

.c0 > span {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  z-index: 1;
}

.c0:focus {
  box-shadow: 0 0 0 3px;
}

.c0:disabled {
  cursor: not-allowed;
}

<button
  className="c0 btn"
/>
`;
